src/frontend/src/components/authenticateBox.ts

```diff
@@ -77,27 +78,39 @@ export const authenticateBox = async (
   // Retry until user has successfully authenticated
   for (;;) {
     const result = await promptAuth();
-    switch (result.tag) {
-      case "ok":
-        setAnchorUsed(result.userNumber);
-        return result;
-      case "err":
-        await displayError({
-          title: result.title,
-          message: result.message,
-          detail: result.detail !== "" ? result.detail : undefined,
-          primaryButton: "Try again",
-        });
-        break;
-      case "canceled":
-        break;
-      default:
-        unreachable(result);
-        break;
+    const { success: loginData } = await handleLoginFlowResult(result);
+
+    if (nonNullish(loginData)) {
+      return loginData;
     }
   }
 };
 
+export const handleLoginFlowResult = async (
```

Comment:
would it make sense for this to leave in `flowResult.ts`?

Reply:
Yes but, see PR description, in a separate PR.